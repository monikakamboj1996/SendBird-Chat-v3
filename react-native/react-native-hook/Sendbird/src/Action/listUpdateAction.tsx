export const ListUpdate = list => {
  return {
    type: 'List_Update',
    list: list,
  };
};
