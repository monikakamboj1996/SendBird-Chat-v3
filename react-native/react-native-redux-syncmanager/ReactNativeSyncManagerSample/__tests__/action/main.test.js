import 'react-native';
import React from 'react';
import Action from '../../src/action/main';

describe('action/chat', () => {
  it('addChannels', () => {
    const result = Action.addChannels([{ url: 'test' }, { url: 'test 2' }]);
    expect(result).toMatchSnapshot();
  });

  it('updateChannels', () => {
    const result = Action.updateChannels([{ url: 'test' }, { url: 'test 2' }]);
    expect(result).toMatchSnapshot();
  });

  it('removeChannels', () => {
    const result = Action.removeChannels([{ url: 'test' }, { url: 'test 2' }]);
    expect(result).toMatchSnapshot();
  });

  it('moveChannels', () => {
    const result = Action.moveChannels([{ url: 'test' }, { url: 'test 2' }]);
    expect(result).toMatchSnapshot();
  });

  it('clearChannels', () => {
    const result = Action.clearChannels();
    expect(result).toMatchSnapshot();
  });
});
