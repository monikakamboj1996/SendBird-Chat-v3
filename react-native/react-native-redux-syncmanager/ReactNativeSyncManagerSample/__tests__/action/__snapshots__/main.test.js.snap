// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action/chat addChannels 1`] = `
Object {
  "payload": Object {
    "channels": Array [
      Object {
        "url": "test",
      },
      Object {
        "url": "test 2",
      },
    ],
  },
  "type": "channel-add",
}
`;

exports[`action/chat clearChannels 1`] = `
Object {
  "type": "channel-clear",
}
`;

exports[`action/chat moveChannels 1`] = `
Object {
  "payload": Object {
    "channels": Array [
      Object {
        "url": "test",
      },
      Object {
        "url": "test 2",
      },
    ],
  },
  "type": "channel-move",
}
`;

exports[`action/chat removeChannels 1`] = `
Object {
  "payload": Object {
    "channels": Array [
      Object {
        "url": "test",
      },
      Object {
        "url": "test 2",
      },
    ],
  },
  "type": "channel-remove",
}
`;

exports[`action/chat updateChannels 1`] = `
Object {
  "payload": Object {
    "channels": Array [
      Object {
        "url": "test",
      },
      Object {
        "url": "test 2",
      },
    ],
  },
  "type": "channel-update",
}
`;
